File: Src/Services/Common/Common/Data/PagedResult.cs
namespace Common.Data
{
    public class PagedResult
    {
        public const int PageSize = 10;
        public required int TotalCount { get; init; }
        public required int TotalPages { get; init; }
        public required int CurrentPage { get; init; }
    }

    public class PagedResult<T> : PagedResult
    {
        public required T Data { get; init; }
    }
}

File: Src/View/MicroShop.Admin/ClientApp/src/modules/shared/models/paged-result.ts
export interface Result {
    status: number;
    statusText: string;
}

export interface DataResult<T> extends Result {
    data: T;
}

export interface PagedResult<T> extends DataResult<T> {
    pager: {
        totalCount: number;
        totalPages: number;
        currentPage: number;
    }
}